.github/workflows/pypi_publish.yml

```diff
@@ -40,8 +53,76 @@ jobs:
           overwrite: true
           file_glob: true
 
-      - name: Publish to PyPI (${{vars.PYPI_PUBLISH_URL}})
-        uses: pypa/gh-action-pypi-publish@v1.12.2
+      - name: Publish to PyPI
+        uses: pypa/gh-action-pypi-publish@v1.10.3
+
+  publish_sdm:
+    name: Publish SDM to DockerHub
+    if: startsWith(github.ref, 'refs/tags/v') || github.event_name == 'workflow_dispatch'
+    runs-on: ubuntu-latest
+    needs: [publish]
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set Version (workflow_dispatch)
+        if: github.event_name == 'workflow_dispatch'
+        run: |
+          echo "Version set to ${{ github.event.inputs.version }}"
+          echo "VERSION=${{ github.event.inputs.version }}" >> $GITHUB_ENV
+
+      - name: Set Version (tag)
+        if: startsWith(github.ref, 'refs/tags/v')
+        run: |
+          echo "Version set to ${{ github.ref_name }}"
+          echo "VERSION=${{ github.ref_name }}" >> $GITHUB_ENV
+
+      # We need to download the build artifact again because the previous job was on a different runner
+      - name: Download Build Artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: Packages-${{ github.run_id }}
+          path: dist
```

Comment:
Adding this step is necessary to ensure that the SDM image has access to the build artifact and can access the correct CDK version